# This patch file was generated by NetBeans IDE
# It uses platform neutral UTF-8 encoding and \n newlines.
--- HEAD
+++ Modified In Working Tree
@@ -41,6 +41,7 @@
 import com.sonyericsson.jenkins.plugins.bfa.graphs.ProjectGraphAction;
 import com.sonyericsson.jenkins.plugins.bfa.model.FailureCause;
 import com.sonyericsson.jenkins.plugins.bfa.model.FailureCauseBuildAction;
+import com.sonyericsson.jenkins.plugins.bfa.model.FailureCauseDisplayData;
 import com.sonyericsson.jenkins.plugins.bfa.model.FailureCauseMatrixBuildAction;
 import com.sonyericsson.jenkins.plugins.bfa.model.FailureReader;
 import com.sonyericsson.jenkins.plugins.bfa.model.FoundFailureCause;
@@ -148,6 +149,17 @@
             FailureCauseBuildAction buildAction = new FailureCauseBuildAction(foundCauseList);
             buildAction.setBuild(build);
             build.addAction(buildAction);
+            
+            final FailureCauseDisplayData data = buildAction.getFailureCauseDisplayData();
+            List<FailureCauseDisplayData> downstreamFailureCauses = data.getDownstreamFailureCauses();
+                    
+            if ( ! downstreamFailureCauses.isEmpty()) {
+                buildLog.println("[BFA] Found downstream Failure causes ...");
+                for (FailureCauseDisplayData displayData : downstreamFailureCauses) {
+                FailureCauseDisplayData.Links links = displayData.getLinks(); 
+                buildLog.println("[BFA] See "+links.getProjectDisplayName()+links.getBuildDisplayName());
+            } 
+            }
             StatisticsLogger.getInstance().log(build, foundCauseListToLog);
         } catch (Exception e) {
             logger.log(Level.SEVERE, "Could not scan build " + build, e);
@@ -199,7 +211,16 @@
                     {build.getFullDisplayName(),
                             String.valueOf(time), });
         }
-        buildLog.println();
+        if (!foundFailureCauseList.isEmpty()) {
+            buildLog.println("[BFA] Found failure cause(s):");
+            for (FoundFailureCause foundCause : foundFailureCauseList) {
+                buildLog.println(foundCause.getName());                
+            }
+            
+        }
+        else {
+            buildLog.println("[BFA] No failure causes found");
+        }
         buildLog.println("[BFA] Done. " + TimeUnit.MILLISECONDS.toSeconds(time) + "s");
         return foundFailureCauseList;
     }

